sycl/source/detail/kernel_impl.cpp

```diff
@@ -57,12 +57,32 @@ kernel_impl::~kernel_impl() {
   }
 }
 
+template <info::kernel param>
+typename info::param_traits<info::kernel, param>::return_type
+get_host_kernel_info() {
+  throw runtime_error(
+      "This kernel info parameter is not supported on HOST device.");
+}
+
+template <> string_class get_host_kernel_info<info::kernel::function_name>() {
+  return "Host kernel";
+}
+
+template <> cl_uint get_host_kernel_info<info::kernel::num_args>() { return 0; }
+
+template <> cl_uint get_host_kernel_info<info::kernel::reference_count>() {
```

Comment:
According to spec:
Returns the reference count of the encapsulated SYCL cl_kernel, if this SYCL kernel is an OpenCL program. Must thrown a invalid_object_error SYCL exception if this SYCL kernel is a host kernel.

Reply:
Thanks, corrected it.